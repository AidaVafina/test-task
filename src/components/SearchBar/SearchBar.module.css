.searchContainer {
    width: 100%;
    margin-bottom: 1rem;
    display: flex;
    justify-content: center;
}

.input {
    width: 100%;
    max-width: 400px;
    padding: 0.5rem 1rem;
    font-size: 1rem;
    border: 2px solid var(--color-border, #ccc);
    border-radius: 4px;
    outline: none;
    transition: border-color 0.2s ease-in-out;
}

.input:focus {
    border-color: var(--color-primary, #3498db);
}
