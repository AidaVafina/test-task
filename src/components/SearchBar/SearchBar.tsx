import React, {type ChangeEvent, useState} from 'react';
import styles from './SearchBar.module.css';
import { useDebounce } from '../../hooks/useDebounce';

interface SearchBarProps {
    onSearch: (query: string) => void;
}

const SearchBar: React.FC<SearchBarProps> = ({ onSearch }) => {
    const [value, setValue] = useState<string>('');
    const debouncedValue = useDebounce(value, 300);

    React.useEffect(() => {
        onSearch(debouncedValue);
    }, [debouncedValue, onSearch]);

    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
        setValue(e.target.value);
    };

    return (
        <div className={styles.searchContainer}>
            <input
                type="text"
                value={value}
                onChange={handleChange}
                className={styles.input}
                placeholder="Поиск по имени пользователя..."
            />
        </div>
    );
};

export default SearchBar;
