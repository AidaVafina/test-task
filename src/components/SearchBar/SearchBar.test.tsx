import { render, fireEvent, act } from '@testing-library/react';
import { describe, it, beforeEach, afterEach, expect, vi } from 'vitest';
import SearchBar from './SearchBar';

describe('SearchBar Component', () => {
    const mockOnSearch = vi.fn();

    beforeEach(() => {
        mockOnSearch.mockClear();
        vi.useFakeTimers();
    });

    afterEach(() => {
        vi.useRealTimers();
    });

    it('should match snapshot', () => {
        const { container } = render(<SearchBar onSearch={mockOnSearch} />);
        expect(container).toMatchSnapshot();
    });

    it('should render search input', () => {
        const { container } = render(<SearchBar onSearch={mockOnSearch} />);
        expect(container.querySelector('input')).toBeInTheDocument();
    });

    it('should call onSearch when input changes', () => {
        const { container } = render(<SearchBar onSearch={mockOnSearch} />);
        const input = container.querySelector('input');
        
        if (input) {
            fireEvent.change(input, { target: { value: 'test' } });

            act(() => {
                vi.advanceTimersByTime(300);
            });
        }

        expect(mockOnSearch).toHaveBeenCalledWith('test');
    });
}); 