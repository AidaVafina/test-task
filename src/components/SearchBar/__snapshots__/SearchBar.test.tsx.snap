// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar Component > should match snapshot 1`] = `
<div>
  <div
    class="_searchContainer_90c5e6"
  >
    <input
      class="_input_90c5e6"
      placeholder="Поиск по имени пользователя..."
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`SearchBar Component should match snapshot 1`] = `
<div>
  <div
    class="searchContainer"
  >
    <input
      class="input"
      placeholder="Поиск по имени пользователя..."
      type="text"
      value=""
    />
  </div>
</div>
`;
